Source/WTF/wtf/text/Base64.h

```diff
@@ -71,11 +67,11 @@ String base64EncodeToString(std::span<const uint8_t>, Base64EncodeMode = Base64E
 WTF_EXPORT_PRIVATE String base64EncodeToStringReturnNullIfOverflow(std::span<const std::byte>, Base64EncodeMode = Base64EncodeMode::Default);
 String base64EncodeToStringReturnNullIfOverflow(const CString&, Base64EncodeMode = Base64EncodeMode::Default);
 
-WTF_EXPORT_PRIVATE std::optional<Vector<uint8_t>> base64Decode(std::span<const std::byte>, Base64DecodeMode = Base64DecodeMode::DefaultIgnorePadding);
-WTF_EXPORT_PRIVATE std::optional<Vector<uint8_t>> base64Decode(StringView, Base64DecodeMode = Base64DecodeMode::DefaultIgnorePadding);
-std::optional<Vector<uint8_t>> base64Decode(std::span<const uint8_t>, Base64DecodeMode = Base64DecodeMode::DefaultIgnorePadding);
+WTF_EXPORT_PRIVATE std::optional<Vector<uint8_t>> base64Decode(std::span<const std::byte>, OptionSet<Base64DecodeOption> = { });
+WTF_EXPORT_PRIVATE std::optional<Vector<uint8_t>> base64Decode(StringView, OptionSet<Base64DecodeOption> = { });
+std::optional<Vector<uint8_t>> base64Decode(std::span<const uint8_t>, OptionSet<Base64DecodeOption> = { });
 
-WTF_EXPORT_PRIVATE String base64DecodeToString(StringView, Base64DecodeMode = Base64DecodeMode::DefaultIgnorePadding);
+WTF_EXPORT_PRIVATE String base64DecodeToString(StringView, OptionSet<Base64DecodeOption> = { });
```

Comment:
These all used to ignore padding by default. Because we’re changing the default, we now have to touch every call site. We could retain the "ignore padding by default" behavior in one of two ways:
1) Have the flag be `ValidatePadding` instead of `IgnorePadding` with a reverse sense.
2) Have the default value be `{ Base64DecodeOption::IgnorePadding }` instead of `{ }`.
We should do one of those. I prefer (1) because it’s better for URL mode. I don’t think separating this out into distinct flags should require changing the default.

Reply:
it felt safer to have the default behavior be to validate padding (i.e. one must opt-out of any/all validation steps), but i dont have a strong opinion so i can definitely change it to be `ValidatePadding` instead